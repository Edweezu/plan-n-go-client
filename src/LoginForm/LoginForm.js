import React from 'react';
import { Link } from 'react-router-dom' 
import TripsContext from '../TripsContext'
import UsersApiService from '../services/users-api-service'
// import TokenService from '../services/token-service'
// import config from '../config'

export default class LoginForm extends React.Component {

    static contextType = TripsContext

    state = {
        error: null
    }

    handleSubmit = (e) => {
        e.preventDefault()
        this.setState({
            error: null
        })

        const { username, password} = e.target

        return UsersApiService.postLogin({
            username: username.value,
            password: password.value
        })
            .then(responseJson => {
               username.value = ''
               password.value = ''
               this.context.loggedIn()
               this.props.onLoginSuccess()
            })
            .catch(responseJson => {
                this.setState({
                    error: responseJson.error
                })
            })
    }


    render () {
        const { error } = this.state
        return (
            <form className='login-form' onSubmit={this.handleSubmit}>
                <div role='alert'>
                    {error && <p className='red'>{error}</p>}
                </div>
                <div className='signup-element'>
                    <div className='signup-label'>
                        <label htmlFor='username'>
                            Username    
                        </label>
                        <span className='astrik'>
                            *
                        </span>
                    </div>
                    <div className='signup-input'>
                        <input id='username' name='username' type='text' placeholder='demo' required/>
                    </div>
                    
                </div>
                <div className='signup-element'>
                    <div className='signup-label'>
                        <label htmlFor='password'>
                            Password    
                        </label>
                        <span className='astrik'>*</span>
                    </div>
                    <div className='signup-input'>
                        <input id='password' name='password' type='password' placeholder='Testing123!' required/>
                    </div> 
                </div>
                <div className='signin-button'>
                    <button className="btn btn-sign-in" type='submit'>
                        Sign In 
                    </button>
                </div>
                <div className='login-demo'>
                    <h4>Demo Account</h4>
                    <p>Username: demo</p>
                    <p>Password: Testing123!</p>
                </div>
                <div className='login-form-redirect'>
                    <Link to='/register'>Don't have an account? Create one.</Link>
                </div>
            </form>
        )
    }
}