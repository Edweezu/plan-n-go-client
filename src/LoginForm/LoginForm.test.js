import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import LoginForm from './LoginForm'


describe('LoginForm component', () => {
    it('renders the given component', () => {
        const wrapper = shallow(<LoginForm />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})