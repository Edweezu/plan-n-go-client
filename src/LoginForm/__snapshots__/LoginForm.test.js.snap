// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders the given component 1`] = `
<form
  className="login-form"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="signup-element"
  >
    <div
      className="signup-label"
    >
      <label
        htmlFor="username"
      >
        Username
      </label>
      <span
        className="astrik"
      >
        *
      </span>
    </div>
    <div
      className="signup-input"
    >
      <input
        id="username"
        name="username"
        placeholder="demo"
        required={true}
        type="text"
      />
    </div>
  </div>
  <div
    className="signup-element"
  >
    <div
      className="signup-label"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <span
        className="astrik"
      >
        *
      </span>
    </div>
    <div
      className="signup-input"
    >
      <input
        id="password"
        name="password"
        placeholder="Testing123!"
        required={true}
        type="password"
      />
    </div>
  </div>
  <div
    className="signin-button"
  >
    <button
      className="btn btn-sign-in"
      type="submit"
    >
      Sign In
    </button>
  </div>
  <div
    className="login-demo"
  >
    <h4>
      Demo Account
    </h4>
    <p>
      Username: demo
    </p>
    <p>
      Password: Testing123!
    </p>
  </div>
  <div
    className="login-form-redirect"
  >
    <Link
      to="/register"
    >
      Don't have an account? Create one.
    </Link>
  </div>
</form>
`;
